src/main/java/io/github/bobocodebreskul/server/DispatcherServlet.java

```diff
@@ -143,6 +157,80 @@ private void processRequest(HttpServletRequest req, HttpServletResponse resp) {
     }
   }
 
+  /**
+   * Processes the method parameter.
+   *
+   * @param parameter The method parameter to be processed.
+   * @param req       The HttpServletRequest.
+   * @param resp      The HttpServletResponse.
+   * @return The processed parameter.
+   * @throws RuntimeException If an error occurs during processing.
+   */
+  private Object processMethodParameter(Parameter parameter, HttpServletRequest req,
+      HttpServletResponse resp) {
+    try {
+      log.debug("Processing method parameter: {}", parameter.getName());
+
+      if (isHttpRequest(parameter)) {
+        return req;
+      }
+
+      if (isHttpResponse(parameter)) {
+        return resp;
+      }
+
+      if (parameter.isAnnotationPresent(RequestBody.class)) {
+        validateRequestMethod(req);
+        return getBodyFromRequest(parameter.getType(), req);
+      }
+
+      throw new WebMethodParameterException("Unsupported parameter type: " + parameter.getType());
+
+    } catch (Exception e) {
+      log.error("Error processing method parameter", e);
+      throw new WebMethodParameterException("Error processing method parameter", e);
+    }
+  }
+
+  private boolean isHttpRequest(Parameter parameter) {
+    return HttpServletRequest.class.isAssignableFrom(parameter.getType());
+  }
+
+  private boolean isHttpResponse(Parameter parameter) {
+    return HttpServletResponse.class.isAssignableFrom(parameter.getType());
+  }
+
+  private void validateRequestMethod(HttpServletRequest req) {
+    if (RequestMethod.GET.name().equals(req.getMethod())) {
+      log.error("GET request not allowed for @RequestBody parameter.");
+      throw new WebMethodParameterException("GET http method not support request body");
+    }
+  }
+
+  /**
+   * Retrieves the request body for the given type.
+   *
+   * @param bodyType The type of the expected request body.
+   * @param req      The HttpServletRequest.
+   * @return The request body object.
+   * @throws RuntimeException If an error occurs while retrieving or parsing the request body.
+   */
+  private Object getBodyFromRequest(Class<?> bodyType, HttpServletRequest req) {
+    try {
+      log.debug("Retrieving request body for type: {}", bodyType.getSimpleName());
+
+      String body = req.getReader()
+          .lines()
+          .collect(Collectors.joining(System.lineSeparator()));
+
+      return mapper.readValue(body, bodyType);
+
+    } catch (IOException e) {
+      log.error("Error reading request body", e);
+      throw new WebMethodParameterException("Error reading request body", e);
+    }
+  }
```

Comment:
Shouldn't we validate somewhere that 'application/json' type is set to the request? I believe we should write understandable error when somebody will try to send XML instead of expected JSON.

Reply:
we send exception inside to save stack trace, so jackson already do this thing, but yes i will try to do more useful massage which help us